### DISCUSSION TITLE
Make fields from new Global Options available in header twig template

### DISCUSSION THREAD

**Discussion by Iain-g** [flyntwp/flynt#475] [Q&A] [answered]
I've managed to add fields to Global Options by duplicating _functions.php_ from _BlockCookieNotice_ and adapting:

`<?php

namespace Flynt\Components\SocialLinks;

use Flynt\Utils\Options;
use Flynt\FieldVariables;
use Timber\Timber;

add_action('wp_footer', function () {
    $context = Timber::get_context();
    Timber::render_string('{{ renderComponent("SocialLinks") }}', $context);
});

Options::addGlobal('SocialLinks', [
    [
      'label' => __('Facebook', 'flynt'),
      'name' => 'SocialFacebook',
      'type' => 'text',
      'instructions' => __('Full link including https://', 'flynt'),
    ],
    [
      'label' => __('Instagram', 'flynt'),
      'name' => 'SocialInstagram',
      'type' => 'text',
      'instructions' => __('Full link including https://', 'flynt'),
    ],
]);`

How do I now make the fields available to my header twig template? A Dump doesn't currently show any social fields available.

**Comment by timohubois** [comment, marked as answer]
Hi @Iain-g the default approach is using the `renderComponent` twig function. What do you mean with header twig template and what exactly do you want to to do? 

If you want to render the component **SocialLinks** with twig you can add `{{ renderComponent("SocialLinks") }}` into any **.twig** file. 

If you want create fields which should be available for multiple components without any layout logic you can take a look on the [sliderOptions.php](https://github.com/flyntwp/flynt/blob/cee63d8b425eed23ac5a0903f5f510d4790eebf1/inc/sliderOptions.php) inside the inc folder. Which will then added to the component `SliderImages` inside its [functions.php](https://github.com/flyntwp/flynt/blob/cee63d8b425eed23ac5a0903f5f510d4790eebf1/Components/SliderImages/functions.php#L9).

**Reply by Iain-g** [reply]
Thanks @timohubois that's done the trick, your first suggestion is what I was after, I've done a timber site before, but not flynt, components is all new to me - loving it! I see now that the answer is even in the add_action code (if you know what you're looking for :D) - Cheers!!
